Topics/Tech_Stacks/Unity_ML_Agents.md

```diff
@@ -0,0 +1,199 @@
+
+# Training
+
+And with that, our setup is almost ready for training. Head to the inspector of our Agent. Observe the behavior parameters. There are 2 values that we should change. Feel free to assign a name to this behaviour if you would like to.
+
+<img src="https://raw.githubusercontent.com/Dismany0/misc_images/main/CSC301%20Assignment%202/Screenshot%202023-11-26%20025032.png" height="500"/>
+
+Our **Vector Observation** should determine the amount of observations our agent is able to make. This should match the amount of observations we gave it in our prior example, which is 3. If your training set up differs, make sure to assign the number of observations correctly.
+
+Next, under the **Actions** section, we want to set the number of Discrete Branches to 1. This is where we are making our observations. Next, we want to set the size of our branch to 3. This should correspond to the number of possible actions our agent is able to make. As our agent is only able to move left, right, and stay still, the size of our action space is 3, as we assigned in the Actions section above.
+
+<img src="https://raw.githubusercontent.com/Dismany0/misc_images/main/CSC301%20Assignment%202/Screenshot%202023-11-26%20031918.png" height="500"/>
+
+The final step is to add a Decision Requester. The only parameter here is how often our agent will make an action. Feel free to play with this value.
+
+With that, we can begin training our agent.
+We can do so by opening a terminal and activating our virtual environment where we previously installed ML-Agents. To begin training, we will run 
+
+```mlagents-learn --run-id="YOUR RUN ID HERE"```
+
+Replacing with your own run ID. This ID could be anything that helps you identify a session of training. 
+
+For additional parameters, run ```mlagents-learn --help``` to view them all.
+
+Annnnnnnnd voila!
+
+<img src="https://github.com/Dismany0/misc_images/blob/main/CSC301%20Assignment%202/ezgif.com-video-to-gif%20(3).gif?raw=true" height="500"/>
+
+You've just learned how to train your first agent in Unity! While the agent might not appear very smart at first, it should eventually learn that moving towards the larger reward is much more rewarding than the smaller one. Over time, you should expect to see the average reward, provided every few minutes in the terminal, to increase.
+
+Feel free to make your own adjustments or create your own training environment. Fun additions you could try to build off of this example can include
+- Movement along a 2d square instead of a line
+- A penalty for taking a long time to reach a reward
+- Additional obstacles your agent must avoid
+
+Feel free to explore both on your own, and with Unity's [official documentation](https://unity-technologies.github.io/ml-agents/ML-Agents-Toolkit-Documentation/) and the [ML-Agents Repo](https://github.com/Unity-Technologies/ml-agents).
```

Comment:
You said feel free 6 times in the page, could look for synonyms to replace some of them

Reply:
haha I realized that too reading through mine another time. I replaced most of them with other phrases to make it a little less jarring to read over and over again. I also reworded the final section with the two links.